### DISCUSSION TITLE
Help Accessing Box-Player Class

### DISCUSSION THREAD

**Discussion by Jfreundlich01** [cwendt94/espn-api#154] [ESPN API] [answered]
Would like to start this off by saying how impressed I am with this API, been looking for something like this for a while, and am having a lot of fun digging around in it. I am relatively new to coding, have some experience with HTML, Java and SQL, but not super familiar with Python. 

I am trying to access the Box Player class. 

Once I import my league and attach it to a variable 'league' this is my code:

teams = league.teams
team = teams[0]

player = team.roster[0]
name = player.name
proj = name.projected_points

print(name, proj)

Not sure what I am doing wrong. Any help is appreciated!

**Comment by cwendt94** [comment, marked as answer]
Hey @Jfreundlich01, I am glad this package has been useful! 

So after you have created a League object the players on a teams roster is from the [Player Class](https://github.com/cwendt94/espn-api/wiki/Player-Class) which just has all of the general information of that player across the season. If you want to get matchup and specific week player data you can call the `league.box_scores()` interface which returns [Box Score Class](https://github.com/cwendt94/espn-api/wiki/Box-Score-Class)

I hope this helps!

**Reply by Jfreundlich01** [reply]
Got it! Thanks for the help!

**Reply by ghost** [reply]
> Hey @Jfreundlich01, I am glad this package has been useful!
> 
> So after you have created a League object the players on a teams roster is from the [Player Class](https://github.com/cwendt94/espn-api/wiki/Player-Class) which just has all of the general information of that player across the season. If you want to get matchup and specific week player data you can call the `league.box_scores()` interface which returns [Box Score Class](https://github.com/cwendt94/espn-api/wiki/Box-Score-Class)
> 
> I hope this helps!

So where in the documentation is "league.box_scores()"? I looked at the wiki and didn't see anything about it. Like how do I get info from all the matchups over the year?

**Reply by cwendt94** [reply]
You can find example of using `box_scores` on the main [Football Page](https://github.com/cwendt94/espn-api/wiki/Football-Intro) and little overview on the Football [League Interface Page](https://github.com/cwendt94/espn-api/wiki/League-Class).

If you want to get info from all the matchups over the year you can loop through each week and call `box_scores()` For example
```python3
# your number of weeks in league
for week in range(16)
   matchup = league.box_scores(week)
```
